pyvista/plotting/tools.py

```diff
@@ -459,23 +460,78 @@ def opacity_transfer_function(mapping, n_colors, interpolate=True,
 
 
 def parse_color(color, opacity=None, default_color=None):
-    """Parse color into a vtk friendly rgb list.
+    """Parse color into a VTK friendly RGB(A) list.
+
+    If ``color`` is a sequence of RGBA floats, the ``opacity`` parameter
+    is ignored.
 
     Values returned will be between 0 and 1.
 
+    Parameters
+    ----------
+    color : str or sequence
+        Either a string, RGB sequence, RGBA sequence, or hex color string.
+        For example:
+
+            * ``'white'``
+            * ``'w'``
+            * ``[1, 1, 1]``
+            * ``[0.5, 1, 0.7, 1]``
+            * ``'#FFFFFF'``
+
+    opacity : float, optional
+        Default opacity of the returned color. Used when ``color`` is
+        not a length 4 RGBA sequence.
+
+    default_color : str or sequence, optional
+        Default color to use when ``color`` is None.  If this value is
+        ``None``, then defaults to the global theme color.  Format is
+        identical to ``color``.
+
+    Returns
+    -------
+    tuple
+        Either a length 3 RGB sequence if opacity is unset, or an RGBA
+        sequence when ``opacity`` is set or the input ``color`` is an
+        RGBA sequence.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> pyvista.parse_color('blue')
+    (0.0, 0.0, 1.0)
+
+    >>> pyvista.parse_color('k')
+    (0.0, 0.0, 0.0)
+
+    >>> pyvista.parse_color('#FFFFFF')
+    (1.0, 1.0, 1.0)
+
+    >>> pyvista.parse_color((0.4, 0.3, 0.4, 1))
+    (0.4, 0.3, 0.4, 1)
+
     """
+    color_valid = True
     if color is None:
         if default_color is None:
             color = pyvista.global_theme.color
         else:
             color = default_color
     if isinstance(color, str):
         color = string_to_rgb(color)
-    elif len(color) == 3:
-        pass
-    elif len(color) == 4:
-        color = color[:3]
+    elif isinstance(color, (Sequence, np.ndarray)):
+        try:
+            color = np.asarray(color, dtype=np.float64)
```

Comment:
Why the explicit conversion to double?

Reply:
That replaces the dtype check. If the conversion fails (non numeric data provided as input), we catch the ValueError and raise the more user friendly ValueError below.

Reply:
Ah, but that might create an unwanted copy I guess...

Reply:
Although it's probably best to have consistent float/double return values maybe?

Reply:
Yeah, the return type is best if type consistent. This can indeed cause potential copies, but we're talking about really small arrays. I wouldn't worry about it.